### ISSUE TITLE
Pyright incorrectly reports arguments are writable in Protocol

### ISSUE THREAD

**Issue by max-muoto** [microsoft/pyright#9040]
When passing in a named tuple or frozen dataclass into a protocol with `Final` annotated fields, Pyright incorrectly reports:

```
"arg_1" is writable in protocol
"arg_2" is writable in protocol  (reportArgumentType)
```

```python
from typing import NamedTuple, Protocol, Final



class MyProtocol(Protocol):
    arg_1: Final[int]
    arg_2: Final[int]


class MyNamedTuple(NamedTuple):
    arg_1: int
    arg_2: int


def foo(arg: MyProtocol): ...

foo(MyNamedTuple(1, 3))
```

The workaround I found I was just using properties for both, which works as expected.

[Pyright Playground](https://pyright-play.net/?pyrightVersion=1.1.381&pythonVersion=3.14&strict=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoByAhhAKYAmAKgK4IA2pANFAArgxgDGYdzAYqiJ0AUKOGc6RAM5SoAWThswHbnQAUSlTwCUALmFRDUIiDQB9AIy6oAlEIDaqGAF0DRk%2BYBM12w6evRCWlZBWIyKloGNTCKGnpSPTdDD0trJyTjUzNvTBQYMXJSYChgMDA1D2sFTS4dawA6RtFS8tCSWMjSNQtmAGZtbSA)

Version: 1.1.381

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.382.
